libraries/AP_ADSB/AP_ADSB.h

```diff
@@ -211,6 +228,7 @@ class AP_ADSB {
 
     } out_state;
 
+    uint8_t num_instances;
```

Comment:
Would be nice to have a better name for this.  Most places we have `num_instances` it really reflects the number of objects we're dealing with.

One option would be to replace `num_instances` with `ADSB_MAX_INSTANCES` in the places we're using it for iteration and to have a different boolean value for where we're checking to see if things are initialised (this calls back to that first comment about the second backend not initialising as the other).